engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreHeartbeatWriteActor.scala

```diff
@@ -40,21 +40,57 @@ case class WorkflowStoreHeartbeatWriteActor(workflowStoreAccess: WorkflowStoreAc
     *
     * @return the number of elements processed
     */
-  override protected def process(data: NonEmptyVector[(WorkflowId, OffsetDateTime)]): Future[Int] = instrumentedProcess {
-    val heartbeatDateTime = OffsetDateTime.now()
-    val processFuture = workflowStoreAccess.writeWorkflowHeartbeats(data, heartbeatDateTime)
-    processFuture transform {
-      // Track the `Try`, and then return the original `Try`. Similar to `andThen` but doesn't swallow exceptions.
-      _ <| trackRepeatedFailures(heartbeatDateTime, data.length)
+  override protected def process(data: NonEmptyVector[WorkflowStoreWriteHeartbeatCommand]): Future[Int] = instrumentedProcess {
+    val now = OffsetDateTime.now()
+
+    val warnDuration = (failureShutdownDuration + workflowHeartbeatConfig.heartbeatInterval) / 2
```

Comment:
Curious what your intent is here. Default in minutes would be `(5 + 2) / 2` or 3.5 minutes. I think I would have initially chosen `heartbeatInterval * 2`, but not sure what I would have done to make sure that's less than `failureShutdownDuration`. 🤔

Reply:
Right, this is certainly more than the heartbeat interval and certainly less than the failure shutdown duration, but not necessarily the "best" choice of value. Suggestions welcome. 🙂

Reply:
We expect heartbeats to be more-or-less realtime, right? I kind of like it as it is now, since it tells us when we're halfway to the failure point.

Reply:
> halfway to the failure point

That's the rationale for this calculation that I was looking for. 👍